python/servo/mutation/test.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2013 The Servo Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import fileinput
+import re
+import subprocess
+import sys
+import os
+import random
+DEVNULL = open(os.devnull, 'wb')
+
+
+def mutate_random_line(file_name):
+    line_numbers = []
+    for line in fileinput.input(file_name):
+        if re.search(r'\s&&\s', line):
+            line_numbers.append(fileinput.lineno())
+    if len(line_numbers) == 0:
+        return -1
+    else:
+        mutation_line_number = line_numbers[random.randint(0, len(line_numbers) - 1)]
+        for line in fileinput.input(file_name, inplace=True):
+            if fileinput.lineno() == mutation_line_number:
+                line = re.sub(r'\s&&\s', ' || ', line)
+            print line.rstrip()
+        return mutation_line_number
+
+
+def mutation_test(file_name, tests):
+    local_changes_present = subprocess.call('git diff --quiet {0}'.format(file_name), shell=True)
```

Comment:
Stylistically, I would prefer to see this whole local changes check broken out into a separate function which exits as failure with a helpful message if local changes were found. Then just call that here instead of having all the actual mutation test logic way down in the `else`. I'd call it `check_local_changes`; you might have a better name.

Reply:
Can you please clarify what you mean by 'separate function exits as failure with a helpful message if local changes were found'?
Does it mean we should exit the program using sys.exit(1)  Or do we just print the error message as we are doing now and proceed to next file mutation? (currently, it skips the file which has local changes)